### ISSUE TITLE
Minutely cronjob fails after swapping out workers with new ASG

### ISSUE THREAD

**Issue by hobbsh** [aws/amazon-vpc-cni-k8s#155]
We use blue/green worker groups to update our AMIs for patching and other fixes. I've noticed that when pods are evicted during the draining, that the CNI starts having problems and the error rate of the cronjob skyrockets unless I delete and re-add it. I see these errors on multiple nodes and not consistently. The job succeeds many times between these errors. This is seemingly a dupe of #59 but I am running v1.1.0 here which supposedly fixed this issue? Our cronjob is not totally broken but our failed cronjob alert is flappy as hell now.

With `kubectl get events`, I see these errors:

> kubelet, ip-10-1-60-40.us-west-2.compute.internal    Failed create pod sandbox: rpc error: code = Unknown desc = NetworkPlugin cni failed to set up pod "REDACTED-1533940200-56dnc_lumo" network: add command: failed to setup network: setup NS network: failed to setup NS network: failed to Statfs "/proc/1577/ns/net": no such file or directory

> kubelet, ip-10-1-60-40.us-west-2.compute.internal    error killing pod: failed to "KillPodSandbox" for "7b36ece3-9ced-11e8-ad95-02af9b42ed2e" with KillPodSandboxError: "rpc error: code = Unknown desc = NetworkPlugin cni failed to teardown pod \"REDACTED-1533940440-z5rns_lumo\" network: rpc error: code = Unavailable desc = grpc: the connection is unavailable"

**Comment by hobbsh** [comment]
@liwenwu-amazon I should be able to replicate this again and can send you the support tarball afterwards. Thanks

**Comment by hobbsh** [comment]
I have tried to replicate this but have not been able to today. @tomweston [on slack](https://kubernetes.slack.com/archives/C09NXKJKA/p1534275684000100) seems to be having the same issue though. I will continue to try to replicate but hopefully his support logs can identify the issue.

**Comment by lstoll** [comment]
We're seeing something similar - repro steps
* Scale node ASG to 0
* Create a pod directly in API, it'll sit in pending
* Scale node ASG up
* Pod will end up in Error state, with events showing items like:
```
  Normal   SandboxChanged          11m (x10 over 11m)  kubelet, ip-10-0-124-11.ec2.internal  Pod sandbox changed, it will be killed and re-created.
  Warning  FailedCreatePodSandBox  11m (x11 over 11m)  kubelet, ip-10-0-124-11.ec2.internal  Failed create pod sandbox: rpc error: code = Unknown desc = NetworkPlugin cni failed to set up pod "test-pod" network: rpc error: code = Unavailable desc = grpc: the connection is unavailable
```

Is there anything blocking pods getting scheduled until the daemon is started and listening? It seems like there's a window where pods can be scheduled before the daemon is listening on the grpc socket, which would cause the cni plugin to fail out like this.

Let me know if that debug bundle would be useful.

**Comment by liwenwu-amazon** [comment]
@lstoll  thank you for providing the repo steps.  
One question:
Does these CronJob Pods eventually get IPs after new node from ASG joins and ipamD on these new nodes finish setting up its initial IP warm Pool?  
thanks

**Comment by liwenwu-amazon** [comment]
Also, when "kubeleet" get error (e.g. network: rpc error ...), it should retry creating "test-pod".  So eventually "test-pod" should get an IP from ipamD.  If this is NOT the case, can you run `/opt/cni/bin/aws-cni-support.sh` and send me `/var/log/aws-routed-eni/aws-cni-support.tar.gz`  thanks

**Comment by liwenwu-amazon** [comment]
@lstoll @hobbsh  I think I may find out the root cause for your issue.  When we introduced the`WARM_IP_TARGET` feature in v1.1,  we change the behavior of allocating the 1st secondary ENI.  And this new behavior might cause you seeing CronJob NOT getting IP for longer period of time compare to v1.0.
 
Here is the behavior difference:
* in v1.0
    * Right after the 1st Cronjob get scheduled onto the node,  ipamD start allocating the 1st secondary ENI and its IP addresses.  In t2.medium example, ipamD should have 10 IP addresses and 2 enI in its warm pool right away.
   
* in v1.1
    * ipamD will NOT allocate the 1st secondary ENI until all IPs of primary interface are assigned to running Pod.  In t2.medium example, 5 running CronJob Pod.  
    * ipamD eni allocation task runs every 5 second.  In t2 medium example, if there are less than 5 running CronJob pod, ipamD will skip allocating a new eni
    * ipamD also have a `ip address cooling period` that it will not give out an IP if it is used by another Pod less than 30 sec ago.
    * if a CronJob runs for only 1 second and finishes. ipamD put this IP address in `cooling period` for 30 sec.  At time when ipamD eni task runs(after 5 sec interval), it will NOT find 5 running pods and will skip allocating new ENI.
    * Once after ipamD allocates the 1st secondary ENI, it behaves same way as v1.0

**Comment by lyn716** [comment]
@liwenwu-amazon After executed `sudo /opt/cni/bin/aws-cni-support.sh`, 
I got `curl: (7) Failed to connect to localhost port 61678: Connection refused`.

**Comment by liwenwu-amazon** [comment]
This should be closed with #169 .  Please re-open it if this is still an issue
